starlette/datastructures.py

```diff
@@ -585,6 +585,16 @@ def __repr__(self) -> str:
 
 
 class MutableHeaders(Headers):
+
+    def __init__(
+        self,
+        headers: typing.Optional[typing.Mapping[str, str]] = None,
+        raw: typing.Optional[typing.List[typing.Tuple[bytes, bytes]]] = None,
+        scope: typing.Optional[typing.Mapping[str, typing.Any]] = None,
+    ) -> None:
+        super().__init__(headers, raw, scope)
+        self._list = list(self._list)
```

Comment:
I think doing this in the parent class is good enough.

Reply:
Also a test case covering the issue is wanted.

Reply:
Feel free to update the branch, I’m on my phone and AFK the rest of the day